src/main/java/seedu/address/logic/parser/DisplayCommandParser.java

```diff
@@ -13,21 +16,81 @@
  */
 public class DisplayCommandParser implements Parser<DisplayCommand> {
 
+    public static final String MESSAGE_INDEX_NOT_PARSED = "The index entered is not within the allowed range. "
+            + "The index cannot be bigger than 2147483647 (i.e., MAX_VALUE).";
+
     /**
      * Parses the given {@code String} of argument and returns a DisplayCommand object for execution.
      * @throws ParseException if the user input does not conform the expected format
      */
     public DisplayCommand parse(String args) throws ParseException {
         requireNonNull(args);
         ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(args);
+        List<String> argList = argMultimap.getAllValues(EMPTY_PREFIX);
 
         Index index;
         try {
+            tryParse(argList);
             index = ParserUtil.parseIndex(argMultimap.getPreamble());
         } catch (IllegalValueException ive) {
             throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, DisplayCommand.MESSAGE_USAGE), ive);
+        } catch (NumberFormatException e) {
+            throw new ParseException(String.format(MESSAGE_INDEX_NOT_PARSED), e);
         }
-
         return new DisplayCommand(index);
     }
+
+    /**
+     * Extracts the last {@code argument} from {@code List<String> argList}.
+     */
+    private String extractLastArgument(List<String> argList) {
+        String last = argList.get(argList.size() - 1); // category is the first argument
+        return last;
+    }
+
+    /**
+     * Checks whether the parameter that has been input is a whole number (can also be a whole number greater
+     * than Integer.MAX_VALUE
+     *
+     * @param arg the parameter that has been entered by the user
+     * @return
+     */
+    public boolean isWholeNumber(String arg) {
+        try {
+            for (char c : arg.toCharArray()) {
+                Integer.parseInt(String.valueOf(c));
+            }
+        } catch (NumberFormatException e) {
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * Checks if there are any parameters input by the user
+     * @param list The list that stores the parameters
+     * @return a boolean indicating whether there is a parameter input or not
+     */
+    public boolean hasParameterValue(List<String> list) {
+        for (String s: list) {
+            if (!s.equals("")) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Tries to parse the input of the user to get the integer index
+     * @param argList List that stores the user input
+     */
+    public void tryParse(List<String> argList) {
```

Comment:
this method should throw a NumberFormatException right? I think it still can get caught without explicitly stating it throws. But I think it would be good to still write it so that when pple look at it, they know NumberFormatException comes from in line 37

Reply:
Also i think u can rename it to tryParseIndex, so pple know u trying to parse the index. Otherwise they wont know wat its going to parse just based off the method name. Just some code quality stuff

Reply:
Okay but its fine to catch it at line 37 right ? I'll make tryParseIndex throw the exception

Reply:
yea its fine to catch it at line 37, was just saying to write `throws NumberFormatException` for the tryparse method

Reply:
yep can i edited it now